Automattic/woocommerce-payments
# Avoid hidden overflow on the ellipsis menu from the task list

Fixes #4470 

#### Changes proposed in this Pull Request

The ellipsis menu uses `position: absolute` in newer versions of `@wordpress/components` (as opposed to `position: fixed` in the v11.1.5 we bundle), and the ExperimentalList from `@woocommerce/experimental` defines an `overflow: hidden` property on each ListItem. The consequence is that the ellipsis menu for a task cannot be properly seen in that case, and we cannot dismiss it.

I opened an issue in the WooCommerce repo: https://github.com/woocommerce/woocommerce/issues/34780 because I think it should be fixed there for all consumers. It used to work fine, but the experimental list package is now not fully compatible with the latest ellipsis menu provided by `@wordpress/components` (not sure from which version it started to go wrong).

Still, I thought of opening this PR with two small overrides on CSS classes so we can fix it immediately in our plugin. Note that I added a new CSS class `wcpay-task-list` to ensure these overrides don't leak in undesired places of the plugin.
If a fix is made in the WooCommerce repo, we can remove this code provided we upgrade to the corresponding version of the  `@woocommerce/experimental` package first.

The reasoning behind the changes are nicely explained here, for reference: https://front-back.com/how-to-make-absolute-positioned-elements-overlap-their-overflow-hidden-parent/

After the changes:
![Sep-22-2022 15-30-00](https://user-images.githubusercontent.com/28830738/191763855-be605a21-e5cc-426a-bd33-30dee33a3dd3.gif)

#### Testing instructions

For testing, you may manually duplicate the "disputes tasks"
* Make sure you have at least one dispute waiting for a response
* To have 2 tasks next to each other vertically for testing purposes, you can manually duplicate this bit of code (beware of changing the `key` field, it's supposed to be unique to be shown): https://github.com/Automattic/woocommerce-payments/blob/0b2030b7a99210461f5a3a35b97f9400a3a1654c/client/overview/task-list/tasks.js#L97-L124

With Gutenberg plugin installed and active:

* Visit Payments > Overview
   * On `develop`, you cannot properly see the ellipsis menu for either of the tasks
   * On this branch, you can see it properly and dismiss the task, for both tasks (you can also check in a mobile view)

Without Gutenberg plugin:
* Visit Payments > Overview
   * On `develop`, you can see the ellipsis menu properly and dismiss the task, for both tasks (you can also check in a mobile view)
   * On this branch, you can also see it properly and dismiss the task, for both tasks (you can also check in a mobile view)
-------------------


**Post merge**
